typescript/phoenix-sdk/examples/simpleMarketMaker.ts

```diff
@@ -1,98 +1,98 @@
-    trader.publicKey,
-    marketPubkey.toString()
+  // If the trader is a new maker (has not placed limit orders previously), you will need to create associated token accounts for the base and quote tokens, and claim a maker seat on the market.
+  // This function creates a bundle of new instructions that includes:
+  // - Create associated token accounts for base and quote tokens, if needed
+  // - Claim a maker seat on the market, if needed
+  const setupNewMakerIxs = await PhoenixSdk.getMakerSetupInstructionsForMarket(
+    client,
+    marketPubkey,
+    marketData,
+    trader.publicKey
   );
-  const seat = client.getSeatKey(trader.publicKey, marketPubkey.toString());
-  const logAuthority = client.getLogAuthority();
-
-  // create account object for each instruction
-  const placeAccounts: Phoenix.PlaceLimitOrderInstructionAccounts = {
-    phoenixProgram: Phoenix.PROGRAM_ID,
-    logAuthority: logAuthority,
-    market: marketPubkey,
-    trader: trader.publicKey,
-    seat: seat,
-    baseAccount: baseAccount,
-    quoteAccount: quoteAccount,
-    quoteVault: marketData.header.quoteParams.vaultKey,
-    baseVault: marketData.header.baseParams.vaultKey,
-    tokenProgram: TOKEN_PROGRAM_ID,
-  };
+  if (setupNewMakerIxs.length !== 0) {
+    const setupTx = new Transaction().add(...setupNewMakerIxs);
+    const setupTxId = await client.connection.sendTransaction(
+      setupTx,
+      [trader],
+      {
+        skipPreflight: true,
+      }
+    );
+    await client.connection.confirmTransaction(setupTxId, "confirmed");
+    console.log(
+      `Setup Tx Link: https://beta.solscan.io/tx/${setupTxId}?cluster=devnet`
+    );
+  }
 
-  //unused in this script, but example of accounts needed for withdraw
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const withdrawAccounts: Phoenix.WithdrawFundsInstructionAccounts = {
-    phoenixProgram: Phoenix.PROGRAM_ID,
-    logAuthority: logAuthority,
-    market: marketPubkey,
-    trader: trader.publicKey,
-    baseAccount: baseAccount,
-    quoteAccount: quoteAccount,
-    baseVault: marketData.header.baseParams.vaultKey,
-    quoteVault: marketData.header.quoteParams.vaultKey,
-    tokenProgram: TOKEN_PROGRAM_ID,
-  };
+  // To place a limit order, you will need base and quote tokens. For this devnet example, we mint the base and quote tokens from a token facuet.
+  // To trade on mainnet, please supply your own base and quote tokens.
+  const airdropSplIxs = await airdropSplTokensForMarketIxs(
+    client,
+    marketData,
+    trader.publicKey
+  );
+  const airdropSplTx = new Transaction().add(...airdropSplIxs);
+  const airdropTxId = await client.connection.sendTransaction(
+    airdropSplTx,
+    [trader],
+    {
+      skipPreflight: true,
+    }
+  );
+  await client.connection.confirmTransaction(airdropTxId, "confirmed");
+  console.log(
+    `Airdrop Tx Link: https://beta.solscan.io/tx/${airdropTxId}?cluster=devnet`
+  );
```

Comment:
does the phoenix client tell us if it's devnet or not? if so, we can wrap this in `if (client.networkIsDevnet())`

Reply:
I don't believe we do; though would not be too much work to add it.